distributed/lock.py

```diff
@@ -93,28 +45,20 @@ class Lock:
     >>> lock.release()  # doctest: +SKIP
     """
 
-    def __init__(self, name=None, client=None):
-        self._client = client
```

Comment:
nit: It would be nice if we still supported the `client` keyword for backward compatibility. IIUC, passing `client=foo_client` would basically be the same as passing `scheduler_rpc=foo_client.scheduler`.

Reply:
`scheduler_rpc` is just used for tests to simulate flaky connections. client isn't and hasn't been used anywhere and I doubt any user will do so.

I'm fine with not going through a deprecation cycle considering that we're very bad at removing deprecation messages

Reply:
I grepped github and actually found a couple of usages where people provided a client... I'll add it with a deprecation warning